mitmproxy/flow.py

```diff
@@ -226,9 +232,8 @@ def resume(self):
         if not self.intercepted:
             return
         self.intercepted = False
-        # If a flow is intercepted and then duplicated, the duplicated one is not taken.
-        if self.reply.state == "taken":
-            self.reply.commit()
+        if self._resume_event is not None:
+            self._resume_event.set()
```

Comment:
There is a race now where we wait indefinitely if we call `intercept()`, then `resume()`, and only then `wait_for_resume()`, no?

Reply:
This is specifically tested in `test/mitmproxy/test_http.py::test_wait_for_resume`. `_resume_event` is only set if `wait_for_resume` gets called, but `wait_for_resume` won't create `_resume_event` nor wait if `intercepted` is False. (You can also see where this would break if these get called from different threads...)

Reply:
Ah true - `wait_for_resume` exits early if intercept is false. Thanks!